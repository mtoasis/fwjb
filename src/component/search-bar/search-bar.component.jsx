import React, { useState } from 'react'
import FormInput from '../form-input/form-input.component'
import { SearchBarContainer } from './search-bar.styles'
import { importUserlistStart } from '../../redux/user/user.action'
import { connect } from 'react-redux'

const SearchBar = ({ importUserlistStart }) => {

    const [searchTerm, setSearchTerm] = useState("");

    const handleChange = event => {
        const { value } = event.target;
        setSearchTerm(value);
    }

    const handleSubmit = event => {
        event.preventDefault();
        importUserlistStart(searchTerm);
    }

    return (
        <SearchBarContainer>
            <form onSubmit={handleSubmit}>
                <FormInput
                    type="text"
                    value={searchTerm}
                    onChange={handleChange}
                    label="Search skills"                    
                />
            </form>
        </SearchBarContainer>
    )
}

const mapDispatchToProps = dispatch => ({
    importUserlistStart: (searchTerm) => dispatch(importUserlistStart(searchTerm))
});

export default connect(null, mapDispatchToProps)(SearchBar);