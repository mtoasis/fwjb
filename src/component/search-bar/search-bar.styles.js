import styled from 'styled-components'

export const SearchBarContainer = styled.div`
display: flex;
height:35px;
align-items:center;
    form{
        width:100%
    }
    
`

SearchBarContainer.displayName = 'SearchBarContainer'