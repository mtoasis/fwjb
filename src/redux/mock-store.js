import configureMockStore from 'redux-mock-store';

export const mockStore = configureMockStore([])