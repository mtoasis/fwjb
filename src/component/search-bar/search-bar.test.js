import React from "react";
import { shallow, mount } from "enzyme";
import SearchBar from "./search-bar.component";
import { mockStore } from "../../redux/mock-store";
import { Provider } from "react-redux";
import * as Search from './search-bar.component';

describe("Search-bar component", () => {
  let wrapper;
  let mockImportUserlistStart;
  let store;

  beforeEach(() => {
    mockImportUserlistStart = jest.fn();
    store = mockStore();
    wrapper = mount(
      <Provider store={store}>
        <SearchBar />
      </Provider>
    );

  });

  it("should render Search-bar component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle change search term on input change', ()=>{

      const setSearchTerm = jest.fn();
      const newWrapper = mount(
        <Provider store={store}>
          <SearchBar onChange={setSearchTerm}/>
        </Provider>
      );;
      const handleChange = jest.spyOn(React, 'useState');
      handleChange.mockImplementation(searchTerm => [searchTerm, setSearchTerm]);
      newWrapper.find('FormInputContainer').simulate('change');
    // console.log(handleChange.mockImplementation(searchTerm => [searchTerm, setSearchTerm]))
      expect(setSearchTerm).toBeTruthy();
//   }
    //   expect(setSearchTerm).toBeFalsy();

  })
// it('should handle change search term on input change', ()=>{
//     //const mockInput = 'a'
//     //expect(Search.handleChange(mockInput)).toEqual('a')
// })
});
