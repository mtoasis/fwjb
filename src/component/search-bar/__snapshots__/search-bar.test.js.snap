// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search-bar component should render Search-bar component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SearchBar)>
    <SearchBar
      importUserlistStart={[Function]}
    >
      <SearchBarContainer>
        <div
          className="sc-AxhCb XVIGq"
        >
          <form
            onSubmit={[Function]}
          >
            <FormInput
              label="Search skills"
              onChange={[Function]}
              type="text"
              value=""
            >
              <styled.div>
                <div
                  className="sc-AxjAm cZLelL"
                >
                  <FormInputContainer
                    onChange={[Function]}
                    type="text"
                    value=""
                  >
                    <input
                      className="sc-AxirZ fYlda-d"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </FormInputContainer>
                  <FormInputLabelContainer
                    onChange={[Function]}
                    type="text"
                    value=""
                  >
                    <label
                      className="sc-AxiKw etWACp"
                      onChange={[Function]}
                      type="text"
                      value=""
                    >
                      Search skills
                    </label>
                  </FormInputLabelContainer>
                </div>
              </styled.div>
            </FormInput>
          </form>
        </div>
      </SearchBarContainer>
    </SearchBar>
  </Connect(SearchBar)>
</Provider>
`;
